include/deal.II/dofs/dof_accessor.templates.h

```diff
@@ -1420,20 +1353,149 @@ namespace internal
         // function DoFCellAccessor::set_mg_dof_indices(), which is called by
         // internal::DoFHandlerImplementation::Policy::Implementation::distribute_mg_dofs().
         // In the case of new use cases, this assert can be removed.
-        Assert(
-          dim == structdim,
-          ExcMessage(
-            "This function is intended to be used for DoFCellAccessor, i.e., dimension == structdim."));
+        Assert(dim == structdim,
+               ExcMessage("This function is intended to be used for "
+                          "DoFCellAccessor, i.e., dimension == structdim."));
 
         process_dof_indices(
           accessor,
           dof_indices,
           fe_index,
-          MGDoFIndexSetter<dim, spacedim, level_dof_access, structdim>(
-            accessor.get_fe(fe_index), level),
-          [](auto &, auto &) {},
+          MGDoFIndexProcessor<dim, spacedim>(level),
+          [](auto &stored_index, auto &dof_index) { stored_index = dof_index; },
           true);
       }
+
+      template <int spacedim>
+      static types::global_dof_index &
+      get_mg_dof_index(
```

Comment:
Do you need so many functions? I feel like you have the exact same function for dim=1,2,3

Reply:
You are right, we do not need 6 functions for all `structdim <= dim <= 3`, but we only need 3, one for `structdim == dim`, one for `structdim = 1, dim > 1`, and one for `structdim = 2, dim 3`, because these access the variables `objects`, `lines` and `quads`, respectively. I will soon push an update.